distributed/shuffle/_merge.py

```diff
@@ -243,15 +243,13 @@ def _cull_dependencies(
         all input partitions. This method does not require graph
         materialization.
         """
-        deps = defaultdict(set)
+        deps = defaultdict(frozenset)
```

Comment:
I think the defaultdict is no longer necessary either

```suggestion
        deps = {}
```